src/internal/connector/graph/errors.go

```diff
@@ -139,3 +140,50 @@ func isTimeoutErr(err error) bool {
 		return false
 	}
 }
+
+// ErrData is a helper function that extracts ODataError metadata from
+// the error.  If the error is not an ODataError type, returns an empty
+// slice.  The returned value is guaranteed to be an even-length pairing
+// of key, value tuples.
+func ErrData(e error) []any {
+	result := make([]any, 0)
+
+	if e == nil {
+		return result
+	}
+
+	odErr, ok := e.(odataerrors.ODataErrorable)
+	if !ok {
+		return result
+	}
+
+	// Get MainError
+	mainErr := odErr.GetError()
+
+	result = appendIf(result, "odataerror_code", mainErr.GetCode())
+	result = appendIf(result, "odataerror_message", mainErr.GetMessage())
+	result = appendIf(result, "odataerror_target", mainErr.GetTarget())
+
+	for i, d := range mainErr.GetDetails() {
+		pfx := fmt.Sprintf("odataerror_details_%d_", i)
```

Comment:
nit: I know it will make the code a bit messier, but maybe it would be good to put the index at the end of the key so it's easier to find and string searches in the logs will be easier as the prefix remains the same (e.x. `odataerror_details_code_1`)

Reply:
point, I'll shape that up.